lib/quicly.c

```diff
@@ -4444,15 +4724,177 @@ static int handle_ping_frame(quicly_conn_t *conn, struct st_quicly_handle_payloa
     return 0;
 }
 
+static struct st_quicly_spare_cid_t *pick_spare_cid(quicly_conn_t *conn)
+{
+    struct st_quicly_spare_cid_t *cid = NULL;
+    /* find the CID with smallest sequence number */
+    for (int i = 0; i < QUICLY_LOCAL_ACTIVE_CONNECTION_ID_LIMIT - 1; i++) {
+        struct st_quicly_spare_cid_t *c = &conn->super.peer.spare_cids[i];
+        if (!c->is_active)
+            continue;
+
+        if (cid == NULL || cid->sequence > c->sequence) {
+            cid = c;
+        }
+    }
+
+    return cid;
+}
+
+/**
+ * copies NEW_CONNECTION_ID information to an internal storage from a received frame
+ */
+static void store_spare_cid(struct st_quicly_spare_cid_t *spare_cid, const quicly_new_connection_id_frame_t *frame)
+{
+    spare_cid->sequence = frame->sequence;
+    set_cid(&spare_cid->cid, frame->cid);
+    memcpy(spare_cid->stateless_reset_token, frame->stateless_reset_token, QUICLY_STATELESS_RESET_TOKEN_LEN);
+    spare_cid->is_active = 1;
+}
+
+/**
+ * compares CIDs, returns zero if identical
+ */
+static int cmp_cid(const quicly_cid_t *l, ptls_iovec_t r)
```

Comment:
Could this be changed to a boolean function that returns true if the two are equal, e.g., `cid_is_equal`?

"cmp" is ambigous. It might imply that it is return a ternary value that can be used for sorting the values being provided, much like `memcmp`. But this function is not designed for such purpose, and IMO there is no need to.

Reply:
👍